axes/middleware.py

```diff
@@ -0,0 +1,29 @@
+from axes.attempts import is_already_locked
+from axes.utils import get_lockout_message
+from django.contrib.auth.backends import ModelBackend
+from django.core.exceptions import PermissionDenied
+
+class DjangoAxesAuthBackend(ModelBackend):
+
+    def authenticate(self, request, username=None, password=None, **kwargs):
+        """
+        Add django-axes handling and add allow adding errors directly to a passed return_context.
+        Will never actually authenticate a user, just blocks locked out uses so don't use this as your only back end.
+        :param request: see ModelBackend.authenticate
+        :param username: see ModelBackend.authenticate
+        :param password: see ModelBackend.authenticate
+        :keyword response_context: context dict that will be returned/used in the response template.
+            NOTE: will overwrite 'error' field in dict
+        :param kwargs: see ModelBackend.authenticate
+        :raises PermissionDenied: if user is already locked out.
+        :return: Nothing, but will update return_context with lockout message if user is locked out.
+        """
+
+        if is_already_locked(request):
+            # locked out, don't try to authenticate, just update return_context and return
+            error_msg = get_lockout_message()
+            return_context = kwargs.get('response_context', {})
+            return_context['error'] = error_msg
```

Comment:
I know this is a bit weird, but I like the option to get messages back from the backend.